### PR TITLE
Update development server options

### LINKED ISSUE #536: Add optional HTTP headers to server
### Problem

Some features (such as required by #459 or #525) require specific HTTP headers to test locally.

### Proposed Solution

- [ ] add `http_headers` and `extra_http_headers` to `LiteBuildConfig`
- [ ] offer default opt-out headers sufficient for #459 and #525

### Additional context

#525 added the [headers it needed](https://github.com/jupyterlite/jupyterlite/pull/525/files#diff-6f16a8748bb80bf5d5de4e82fb5379784df791ee69af72f911ede64a9e6e48daR13), but was not configurable, and  out of scope for that work.

### PR DESCRIPTION
**Pull request by bollwyvl**
## References

- fixes #536
- might help #460
- might help #560

## Code changes

- `jupyter lite serve`
  - [x] adds `http_headers` and `extra_http_headers` (both with empty defaults)
    - [x] tests
  - [x] adds a shutdown method, _should_ help in windows test teardown  
    - [x] tests 
- `doit`
  - [x] `serve:docs:app` runs `jupyter lite serve` with the current `examples` config
  - [x] `serve:py` wrapper for `yarn serve:py` 
  - [x] `serve:js` wrapper for `yarn serve:py`
  - [x] `serve:lab` starts jupyterlab 
  - [x] replaces most "naked" CLI references with `shutil.which`  for windows
- `prettier`
  - [x] don't run prettier on generated `.html` files
- `examples`
  - [x] ~~resolve, including `ipywidgets 7.7.0`~~
    - ~~excited to see what this does with e.g. RTC :tada:~~

## User-facing changes

- n/a

## Backwards-incompatible changes

- n/a

### CODE REVIEW

**Comment by bollwyvl** [comment]
ick:
 
```
ValueError: Requested 'widgetsnbextension~=3.6.0', but widgetsnbextension==3.5.0 is already installed
```

_dang_ i don't like the tilde operator. i'm going to roll back that solve, i guess...

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -87,6 +87,18 @@ changes.
...
+
+A number of development servers can be started, each on separate ports:
+
+- core assets from `./app`:
+  - `doit serve:js`
+  - `doit serve:py`
+- example site it `./build/docs-app`:
+  - `doit serve:docs:app`
+- jupyterlab
+  - `doit serve:lab`
```
Comment (jtpio) [inline CONTRIBUTING.md:100]: Just curious: do we expect contributors to use sock jupyterlab, or is it for convenience?

Comment (bollwyvl) [reply CONTRIBUTING.md:100]: yeah, that and authoring "heavy" docs notebooks like `cli.ipynb`... will add a note above

Comment (bollwyvl) [reply CONTRIBUTING.md:100]: indeed, as a separate item, we should add `jupyterlab-myst`

Comment (bollwyvl) [reply CONTRIBUTING.md:104]: 6e12cbe

**Comment by bollwyvl** [comment]
No further work _planned_ on this... might do some more thinking about, but could probably be a follow-on. the `ipywidgets 7.7.0` might be a better next use of discretionary time, though.

File: CONTRIBUTING.md (hunk 1 above)
Comment (jtpio) [reply CONTRIBUTING.md:104]: Alright sounds good!

**Review by jtpio — APPROVED** [review]
Thanks!

File: dodo.py [hunk 2]
```diff
@@ -1236,7 +1303,7 @@ def deps_to_wheels(all_deps):
             )
         )
         subprocess.check_call(
-            ["pip", "download", "-r", B.RAW_WHEELS_REQS, "--prefer-binary"],
+            [*P.PYM, "pip", "download", "-r", B.RAW_WHEELS_REQS, "--prefer-binary"],
```
Comment (bollwyvl) [inline dodo.py:1306]: this was wrong (it's in `C`): working on a fix along with #565

**Pull request merged**